dist/index.js

```diff
@@ -3304,24 +3307,31 @@ function run() {
             const packageNameList = packageName.split(RegExp("\\s"));
             const rosWorkspaceName = "ros_ws";
             const rosWorkspaceDir = path.join(workspace, rosWorkspaceName);
-            const sourceRosBinaryInstallation = core.getInput("source-ros-binary-installation");
-            const sourceRosBinaryInstallationList = sourceRosBinaryInstallation
-                ? sourceRosBinaryInstallation.split(RegExp("\\s"))
-                : [];
+            const targetRos1Distro = core.getInput("target-ros1-distro");
+            const targetRos2Distro = core.getInput("target-ros2-distro");
             const vcsRepoFileUrlListAsString = core.getInput("vcs-repo-file-url") || "";
             const vcsRepoFileUrlList = vcsRepoFileUrlListAsString.split(RegExp("\\s"));
-            const vcsRepoFileUrlListNonEmpty = vcsRepoFileUrlList.filter(x => x != "");
-            const vcsRepoFileUrlListResolved = vcsRepoFileUrlListNonEmpty.map(x => resolveVcsRepoFileUrl(x));
+            const vcsRepoFileUrlListNonEmpty = vcsRepoFileUrlList.filter((x) => x != "");
+            const vcsRepoFileUrlListResolved = vcsRepoFileUrlListNonEmpty.map((x) => resolveVcsRepoFileUrl(x));
             const coverageIgnorePattern = core.getInput("coverage-ignore-pattern");
             let commandPrefix = "";
-            if (sourceRosBinaryInstallation) {
-                if (process.platform !== "linux") {
-                    core.setFailed("sourcing binary installation is only available on Linux");
+            if (process.platform == "linux") {
```

Comment:
User must always specify a distribution, even if not on Linux. The `if (! && !)` check should not be gated on the platform.